central/development/service/service_impl.go

```diff
@@ -108,15 +110,22 @@ func (s *serviceImpl) ReconciliationStatsByCluster(context.Context, *central.Emp
 }
 
 func (s *serviceImpl) URLHasValidCert(_ context.Context, req *central.URLHasValidCertRequest) (*central.URLHasValidCertResponse, error) {
-	if !strings.HasPrefix(req.GetUrl(), "https://") {
-		return nil, errors.Wrapf(errox.InvalidArgs, "url %q must start with https", req.GetUrl())
+	u, err := url.Parse(req.GetUrl())
+	if err != nil {
+		return nil, errors.Wrapf(errox.InvalidArgs, "invalid url %s", err.Error())
+	}
+
+	if req.CertPEM == "" {
+		_, err = s.client.Get(req.GetUrl())
```

Comment:
Can you please add a comment here how `client.Get` and `verifyProvidedCert` are in fact exercising the same validation logic and **this** is what this is all about, and how this logic is affected by the env variable. TBH it's a mystery to me how `client.Get` gets configured to respect that install-time variable, it looks like a generic 3rd-party library!